src/test/java/sirius/pasta/noodle/compiler/CompilerSpec.groovy

```diff
@@ -37,15 +47,43 @@ class CompilerSpec extends BaseSpecification {
         compile("NoodleExample.AN_INT + NoodleExample.A_LONG_OBJECT").call(new SimpleEnvironment()) == 36L
     }
 
+
     def "accessing fields works"() {
         expect:
         compile("NoodleExample.privateStaticField").call(new SimpleEnvironment()) == "Hello from the other side"
         and:
-        compile("NoodleExample.privateStaticField = 'Hello'; return NoodleExample.privateStaticField;").call(new SimpleEnvironment()) == "Hello"
+        compile("NoodleExample.privateStaticField = 'Hello'; return NoodleExample.privateStaticField;").
+                call(new SimpleEnvironment()) == "Hello"
         and:
         compile("NoodleExample.INSTANCE.privateField").call(new SimpleEnvironment()) == "Hello World"
         and:
-        compile("NoodleExample.INSTANCE.privateField = 'Hello'; return NoodleExample.INSTANCE.privateField;").call(new SimpleEnvironment()) == "Hello"
+        compile("NoodleExample.INSTANCE.privateField = 'Hello'; return NoodleExample.INSTANCE.privateField;").
+                call(new SimpleEnvironment()) == "Hello"
+    }
+
+    def "parsing let/if/for works"() {
+        expect:
+        compile("let x = 5; if (3 < 4) { x = 3; } else { x = 4; }; return x;").call(new SimpleEnvironment()) == 3
+        and:
+        compile("let x = 3; x = 4; return x;").call(new SimpleEnvironment()) == 4
+        and:
+        compile("let sum = 0; for(int x : java.util.Arrays.asList(3, 4)) { sum = sum + x; }; return sum;").
+                call(new SimpleEnvironment()) == 7
+    }
+
+    def "parsing lambdas works"() {
+        expect: "Simple generic type propagation works"
+        compile("let sum = 0; NoodleExample.intStream().forEach(|x| sum = sum + x); return sum;").
```

Comment:
Gibts nen speziellen Grund für die `|x|`-Syntax? In Java würde man ja eher `x ->` erwarten...

Reply:
yep -> is way harder to parse, as it needs an unbound lookahead (we need to consume all parameters commas and braces to eventually check if there is a -> or not - '|' on the other hand can be checked with a lookup of 1 :) - we can still add lambda syntax support later if really needed...

Reply:
wie wäre es mit einen JS `function(x) sum = sum + x` oder Python `lambda x : sum = sum + x` style?

Reply:
The mother of all languages Smalltalk uses this syntax :-P - and rust and a whole bunch of others do as well...